pyproject.toml

```
@@ -1,3 +1,83 @@
+[build-system]
+requires = ["setuptools>=75", "tomli"]
```

```suggestion
requires = ["setuptools>=75.8.1", "tomli"]
```

setuptools < 75.8.1 may cause build failures, hence why we explicitly pin it explicitly in CI (See #7005)

Also, does PL need `tomli` as a build dependency?